app/models/batch_process.rb

```diff
@@ -163,12 +157,22 @@ def configure_parent_object(child_object, parents)
     parents
   end
 
-  def user_update_permission(child_object, parent_object)
+  def user_create_permission(index, admin_set, oid)
+    user = self.user
+    unless user.has_role?(:editor, admin_set)
```

Comment:
we're using `has_role` here instead of `current_ability.can?` because we haven't created a parent object yet.